### ISSUE TITLE
[RFC] Implement a security layer

### ISSUE THREAD

**Issue by gretzky** [react-boilerplate/react-boilerplate#2499]
**Is your feature request related to a problem? Please describe.**
In light of this: [dominictarr/event-stream#116](https://github.com/dominictarr/event-stream/issues/116), we should consider adding a security layer for dependencies.

**Describe the solution you'd like**
There are a few ways we could go about this, but two of them are [Snyk](https://snyk.io/) and [npm-audit-ci](https://www.npmjs.com/package/npm-audit-ci) which would run through Travis. Personally, I've used Snyk as my go-to for this and have enjoyed it.

**Describe alternatives you've considered**
An alternate here would be totally ignoring the situation, but we need to take some responsibility over our dependencies.

**Additional context**
If anyone wants to add their ideas for what we could use, we're open to them.

**Comment by DanielRuf** [comment]
> There are a few ways we could go about this, but two of them are [Snyk](https://snyk.io/) and [npm-audit-ci](https://www.npmjs.com/package/npm-audit-ci) which would run through Travis.

Not sure why the second package is needed (we have `yarn audit` and `npm audit`, also works on CI).

**Comment by gretzky** [comment]
I'm not familiar with the workflow of using `npm audit` with ci (travis specifically), could you elaborate? Like I said I'm only really familiar with snyk

**Comment by julienben** [comment]
@DanielRuf I'd also prefer not adding a new dependency for this but can `npm audit` exit with code 1 based on some criteria? 

@gretzky One thing to consider: Snyk provides free service for OS projects so it will work here but as soon as someone uses it for their own project, they'll have to pay for it. Ideally we would use something that works for our users as well as for us.

**Comment by julienben** [comment]
Seems like we would need to update our minimum npm version to 6.0.1 then? @gretzky How do you feel about that?

**Comment by DanielRuf** [comment]
> Seems like we would need to update our minimum npm version to 6.0.1 then? @gretzky How do you feel about that?

You can do that on CI with postinstall / script entries. Also users should have always the latest stable version of npm.

**Comment by gretzky** [comment]
I don't see a problem updating the minimum npm version, like @DanielRuf it is best practice to keep up w/ latest npm

**Comment by julienben** [comment]
Yeah fair enough. Now that Node LTS is Dubnium/10 it makes even more sense.

**Comment by DanielRuf** [comment]
Ti be exact, active LTS are 6, 8 and 10 ;-)

**Comment by DanielRuf** [comment]
To be exact, active LTS are 6, 8 and 10 ;-)

**Comment by julienben** [comment]
Aye. That's correct. So is it fair of us to require users to be on 10?

Or even just to ask them to update to latest npm (no matter which node version they have)?

**Comment by DanielRuf** [comment]
LTS releases probably already ship with 6.0+ but so far updating / upgrading npm is highly recommended.

**Comment by gretzky** [comment]
^ Agreed, and we can only do so much with others' companies. imo as long as _we_ are doing our due diligence then the rest is up to users.

**Comment by jamesdh** [comment]
I'm not sure there is a need to implement this at all. GitHub provides Dependabot which would do this for you: https://github.com/marketplace/dependabot-preview.

I believe this would also supersede #2673

**Comment by justingreenberg** [comment]
@gretzky @julienben @jamesdh FYI i enrolled the react-boilerplate repo in GitHub's depandabot beta preview (members can [see alerts here](https://github.com/react-boilerplate/react-boilerplate/network/alerts)) and turned on automatic security fixes... however i've been using this feature privately for some time, and it is still very unstable and rarely opens PRs correctly. the integration does display prominent warnings for CVE on the repo homepage:

![image](https://user-images.githubusercontent.com/1539088/63375540-fca9c180-c359-11e9-9337-de3f96c218e6.png)

which is nice, but it's been working on a PR to upgrade lodash since yesterday 😕

**Comment by DanielRuf** [comment]
The yellow banner is an old feature. Only dependabot is really new. Does their status page show issues?

**Comment by justingreenberg** [comment]
yes, here are some screenshots of alert page:

![image](https://user-images.githubusercontent.com/1539088/63377655-7e9be980-c35e-11e9-8963-4cff37e02a5f.png)

and when you click through to an alert (see CVE info at bottom):

![image](https://user-images.githubusercontent.com/1539088/63377689-8e1b3280-c35e-11e9-9138-8bf10f9a231f.png)

**Comment by jamesdh** [comment]
I also saw the lodash warning but did successfully get the PR in one of my repos. I had the opposite problem in that it opened too many PR's since they can't be bundled like Renovate allows, so I reduced it to just security updates also and scheduled it to just weekly.

**Comment by jamesdh** [comment]
@justingreenberg did you try going to the dependabot dashboard and "bumping" the configuration? e.g. https://app.dependabot.com/accounts/react-boilerplate ?

**Comment by justingreenberg** [comment]
@jamesdh thank you for your guidance, seems to be working for dependencies but the security patch for lodash is still "stuck", if it doesn't bump free i'll dismiss and update manually—hopefully this is resolved going forward
